tests/driver_sdcard_spi/main.c

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (C) 2016 Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the sd-card spi driver
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * @}
+ */
+#include "shell.h"
+#include "sdcard_spi.h"
+#include "sdcard_spi_internal.h"
+#include "sdcard_spi_params.h"
+#include "fmt.h"
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+/* independent of what you specify in a r/w cmd this is the maximum number of blocks read at once.
+   If you call read with a bigger blockcount the read is performed in chunks*/
+#define MAX_BLOCKS_IN_BUFFER 4
+#define BLOCK_PRINT_BYTES_PER_LINE 16
+#define FIRST_PRINTABLE_ASCII_CHAR 0x20
+#define ASCII_UNPRINTABLE_REPLACEMENT "."
+
+/* this is provided by the sdcard_spi driver 
+ * see /home/michel/Projekte/fatfs/RIOT/sys/auto_init/storage/auto_init_sdcard_spi.c */
```

Comment:
this path might be hard to follow for people that don't have access to your computer :-)